convex/lib/withZod.ts

```diff
@@ -10,143 +10,63 @@ import {
 import { Id, TableNames } from "../_generated/dataModel";
 
 /**
- * Zod helper for a Convex Id, used for validation.
+ * Create a validator for a Convex `Id`.
```

Comment:
I copied this from the convex monorepo, so lmk if we don't want all these changes